backend/tests/wallet_tests/services/offchain/test_p2m_payment.py

```diff
@@ -147,7 +164,13 @@ def test_approve_payment_success_without_recipient_signature(mock_method):
     mock_method(
         context.get().offchain_client,
         "send_request",
-        will_return=generate_success_init_charge_command_response_object(),
+        will_return=generate_success_init_charge_command_response_object(recipient_signature=None),
+    )
+
+    mock_method(
+        utils,
+        "submit_p2m_txn",
+        will_return=generate_success_p2m_txn_result()
     )
 
     payment_service.approve_payment(user.account_id, REFERENCE_ID)
```

Comment:
[nit] (of the assertion below) - I think that `payment_model.recipient_signature is not None` is more readable and accurate